api/src/main/java/org/infinispan/api/async/AsyncCache.java

```diff
@@ -45,28 +47,105 @@
     */
    CompletionStage<V> get(K key);
 
+   /**
+    * Get the entry of the Key if such exists
+    *
+    * @param key
+    * @return the entry
+    */
+   CompletionStage<CacheEntry<K, V>> getEntry(K key);
+
+   /**
+    * Insert the key/value if such key does not exist
+    *
+    * @param key
+    * @param value
+    * @return the previous value if present
+    */
+   default CompletionStage<V> putIfAbsent(K key, V value) {
+      return putIfAbsent(key, value, CacheEntryExpiration.IMMORTAL);
+   }
+
+   /**
+    * Insert the key/value if such key does not exist
+    *
+    * @param key
+    * @param value
+    * @param expiration
+    * @return the previous value if present
+    */
+   CompletionStage<V> putIfAbsent(K key, V value, CacheEntryExpiration expiration);
+
    /**
     * Insert the key/value if such key does not exist
     *
+    * @param key
+    * @param value
+    * @return
+    */
+   default CompletionStage<Boolean> setIfAbsent(K key, V value) {
+      return setIfAbsent(key, value, CacheEntryExpiration.IMMORTAL);
+   }
+
+   /**
+    * Insert the key/value if such key does not exist
+    *
+    * @param key
+    * @param value
+    * @param expiration
+    * @return Void
+    */
+   CompletionStage<Boolean> setIfAbsent(K key, V value, CacheEntryExpiration expiration);
+
+   /**
     * @param key
     * @param value
     * @return Void
     */
-   CompletionStage<Boolean> putIfAbsent(K key, V value);
+   default CompletionStage<V> put(K key, V value) {
+      return put(key, value, CacheEntryExpiration.IMMORTAL);
+   }
 
    /**
     * @param key
     * @param value
+    * @param expiration
     * @return Void
     */
-   CompletionStage<V> put(K key, V value);
+   CompletionStage<V> put(K key, V value, CacheEntryExpiration expiration);
+
+   /**
+    * @param key
+    * @param value
+    * @return
+    */
+   default CompletionStage<Void> set(K key, V value) {
+      return set(key, value, CacheEntryExpiration.IMMORTAL);
+   }
+
+   /**
+    * @param key
+    * @param value
+    * @param expiration
+    * @return
+    */
+   CompletionStage<Void> set(K key, V value, CacheEntryExpiration expiration);
+
+   /**
+    * @param key
+    * @param value
+    * @return
+    */
+   default CompletionStage<Void> replace(K key, V value, CacheEntryVersion version) {
+      return replace(key, value, CacheEntryExpiration.IMMORTAL, version);
+   }
 
    /**
     * @param key
     * @param value
     * @return
     */
-   CompletionStage<Void> set(K key, V value);
+   CompletionStage<Void> replace(K key, V value, CacheEntryExpiration expiration, CacheEntryVersion version);
```

Comment:
shouldn't the replace return `boolean`? or `CacheEntryVersion` to avoid an extra request

Reply:
actually, it looks like the only API with the `replace` method.

Reply:
All APIs now have `replace` and `getOrReplace`